collector/analyzer/pgftmetricanalyzer/pgft_analyzer.go

```diff
@@ -110,14 +110,23 @@ func (a *PgftMetricAnalyzer) generateSwitchPgft(event *model.KindlingEvent) (*mo
 
 func (a *PgftMetricAnalyzer) getSwitchLabels(event *model.KindlingEvent) (*model.AttributeMap, error) {
 
+	labels := model.NewAttributeMap()
 	ctx := event.GetCtx()
+	if ctx == nil {
+		return labels, nil
+	}
+
```

Comment:
It's typically that we return an error when the return value doesn't contain useful information. As the metrics are meaningless if the labels are empty, we better return an error and print a debug log outside this method to trace the data, which has been done in method `ConsumeEvent` (line 52).